src/Chapter3.hs

```diff
@@ -480,6 +502,41 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+{- ORMOLU_ENABLE -}
+data Castle = MkCastle String
+
+data Wall = BrickWall
+
+data Facility = Church | Library
+
+data People = Citizen
+
+data House = MkHouse [People]
```

Comment:
This is the correct data type, that can represent what we have in requirements 👍🏼 

A few thoughts:
It is more comfortable to work with numbers instead of the length of the list of void elements. In fact, the `People` type doesn't contain any information at all. So maybe you could think of how to represent only possible situations with the limits that the house has on people. In your case, there are no limits, which could lead to illegal states.
With this representation, this is possible. However, we actually can restrict this on type-level by providing more representative types 🙂 This is actually the greatest power of Haskell and type system 😍 

This pattern is called "Make illegal states unrepresentable", and you can read more about it here:

* https://kowainik.github.io/posts/haskell-mini-patterns#make-illegal-states-unrepresentable